core/src/main/kotlin/com/katanox/tabour/configuration/Registry.kt

```diff
@@ -0,0 +1,7 @@
+package com.katanox.tabour.configuration
+
+interface Registry<K> {
+    suspend fun startConsumption()
+    suspend fun stopConsumption()
+    val key: K
```

Comment:
Question: Why do we have the key as generic, I believe it is in most cases a string pair do you see any use case where we make it an object?

Reply:
It can be an enum which makes sense for type safety